### PR TITLE
Move yolo box to phi

### PR DESCRIPTION
**Pull request by phlrain**
### PR types
Breaking changes

### PR changes
OPs

### Describe

move yolo box to phi

### CODE REVIEW

**Review by MingMingShangTian — APPROVED** [review]
LGTM

File: paddle/phi/kernels/cpu/yolo_box_kernel.cc [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  const int an_num = anchors.size() / 2;
+  int input_size_h = downsample_ratio * h;
+  int input_size_w = downsample_ratio * w;
+
+  const int stride = h * w;
+  const int an_stride = (class_num + 5) * stride;
+
+  DenseTensor anchors_;
+  auto anchors_data =
+      anchors_.mutable_data<int>({an_num * 2}, dev_ctx.GetPlace());
```
Comment (chenwhql) [inline paddle/phi/kernels/cpu/yolo_box_kernel.cc:54]: mutable_data改为Alloc

File: paddle/phi/kernels/cpu/yolo_box_kernel.cc [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  std::copy(anchors.begin(), anchors.end(), anchors_data);
+
+  const T* input_data = input->data<T>();
+  const int* imgsize_data = imgsize->data<int>();
+  T* boxes_data = boxes->mutable_data<T>({n, box_num, 4}, dev_ctx.GetPlace());
```
Comment (chenwhql) [inline paddle/phi/kernels/cpu/yolo_box_kernel.cc:59]: 同上

File: paddle/phi/kernels/cpu/yolo_box_kernel.cc [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  memset(boxes_data, 0, boxes->numel() * sizeof(T));
+  T* scores_data =
+      scores->mutable_data<T>({n, box_num, class_num}, dev_ctx.GetPlace());
```
Comment (chenwhql) [inline paddle/phi/kernels/cpu/yolo_box_kernel.cc:62]: 同上

**Pull request merged**